src/search.js

```diff
@@ -248,13 +326,86 @@ class Search {
         return re;
     }
 
+    $isMultilineSearch(options) {
+        return options.re && /\\r\\n|\\r|\\n/.test(options.re.source) && options.regExp && !options.$isMultiLine;
+    }
+
+    $multiLineForward(session, re, start, last) {
+        var line,
+            chunk = chunkEnd(session, start);
+
+        for (var row = start; row <= last;) {
+            for (var i = 0; i < chunk; i++) {
+                if (row > last)
+                    break;
+                var next = session.getLine(row++);
+                line = line == null ? next : line + "\n" + next;
+            }
+
+            var match = re.exec(line);
+            re.lastIndex = 0;
+            if (match) {
+                var before = line.slice(0, match.index).split("\n");
```

Comment:
Can we rewrite the below 2 lines like
    // Split text before match and the match itself into lines
    const beforeMatch = line.slice(0, match.index).split("\n");
    const matchedText = match[0].split("\n");

Reply:
ACK.